### DISCUSSION TITLE
What code to return when supplier returns 429

### DISCUSSION THREAD

**Discussion by JDMKSZ** [belgif/rest-guide#166] [Polls]
For services which consume an upstream service (e.g. from a supplier) what to return when the upstream service is under heavy load and hence doesn't answer? I guess a timeout is always translated to a 502 "Bad Gateway"? What if the upstream service returns a 429? This would mean that the message limit has exceeded, though not necessarily for our own clients. However, it might be interesting to know for them that there is a heavy load and one should retry later.

**Comment by pvdbosch** [comment]
As general rule:
* 4xx = error caused by client (i.e. 429 = client has exceeded its allotted quota)
* 5xx = error caused by server 

A 429 should not be returned if the service was overloaded by other clients than the one making the call.

About choice between status 5xx codes: they're generally treated in the same way by client application logic. I think it depends on which level of detail is deemed useful rather than confusing for users or operators of the client application.
Interpretation of 502 (Bad Gateway) or 504 (Gateway Timeout) errors does require some understanding of the API's internal logic (i.e. there's some backend service being called and failing). This error information might also be hidden from the client and only logged by the server.
